client/rest/src/main/java/org/elasticsearch/client/HeapBufferedAsyncResponseConsumer.java

```diff
@@ -67,6 +67,13 @@ public HeapBufferedAsyncResponseConsumer(int bufferLimit) {
         this.bufferLimit = bufferLimit;
     }
 
+    /**
+     * Get the limit of the buffer.
+     */
+    public int getBufferLimit() {
```

Comment:
I'd really like to see either the response type changed to `ByteSizeValue`, or the name changed to `getBufferLimitBytes()` (preferably the first one)